### ISSUE TITLE
ramda is missing startsWith function

### ISSUE THREAD

**Issue by franciscotln** [DefinitelyTyped/DefinitelyTyped#17428]
- [ x] I tried using the `@types/ramda#0.0.12` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).

@mdekrey @mrdziuban @sbking

Would you guys please add these type definitions to the index.d.ts? It's missing the function startsWith().

```
startsWith(target: string, list: string): boolean;
startsWith(target: string): (list: string) => boolean;
startsWith<T>(target: T[], list: T[]): boolean;
startsWith<T>(target: T[]): (list: T[]) => boolean;
```

**Comment by franciscotln** [comment]
Same thing for `lastIndexOf`. It's only possible to work with arrays and non-curried function.
These definitions are missing:
```
lastIndexOf<T>(target: T): (list: T[]) => number;
lastIndexOf(target: string, list: string): number;
lastIndexOf(target: string): (list: string) => number;
```

**Comment by samsonkeung** [comment]
@franciscotln Type for startsWith has been added.
For lastIndexOf, it does not support curried version based on the Ramda documentation.

**Comment by tycho01** [comment]
@franciscotln consider trying the typings at [`types/npm-ramda`](https://github.com/types/npm-ramda/). while not merged here yet, they're ahead for typing ramda.

**Comment by KubaJastrz** [comment]
Resolved in #20594. Issue can be closed.
